src/oc/web/actions/user.cljs

```diff
@@ -161,11 +161,12 @@
     (api/auth-with-email auth-link email pswd (partial login-with-email-finish email))
     (dis/dispatch! [:login-with-email])))
 
-(defn login-with-slack [auth-url]
+(defn login-with-slack [auth-url & [state-map]]
   (let [auth-url-with-redirect (user-utils/auth-link-with-state
-                                 (:href auth-url)
-                                 {:team-id "open-company-auth"
-                                  :redirect oc-urls/slack-lander-check})]
+                                (:href auth-url)
+                                (or state-map
+                                    {:team-id "open-company-auth"
+                                     :redirect oc-urls/slack-lander-check}))]
```

Comment:
hmm @cjsauer not sure this is what we should be doing here.
Before Slack auth was always done using these `team-id` and `redirect` now instead you are going to do it only with the `redirect-origin` passed from the caller.
I think you should be merging the 2 maps here, `state-map` should be the second so it can override the default properties if passed.
WYT?

Reply:
@bago2k4 yep, resolved 👍